contracts/System.sol

```diff
@@ -3,38 +3,44 @@ pragma solidity 0.8.4;
 
 import "./interface/IRelayerHub.sol";
 
-contract System {
+abstract contract System {
 
   bool public alreadyInit;
+  bool private s_contractAddrUpdated;
 
+  address internal constant _VALIDATOR_CONTRACT_ADDR = 0x0000000000000000000000000000000000001000;
+  address internal constant _SLASH_CONTRACT_ADDR = 0x0000000000000000000000000000000000001001;
+  address internal constant _SYSTEM_REWARD_ADDR = 0x0000000000000000000000000000000000001002;
+  address internal constant _LIGHT_CLIENT_ADDR = 0x0000000000000000000000000000000000001003;
+  address internal constant _RELAYER_HUB_ADDR = 0x0000000000000000000000000000000000001004;
+  address internal constant _CANDIDATE_HUB_ADDR = 0x0000000000000000000000000000000000001005;
+  address internal constant _GOV_HUB_ADDR = 0x0000000000000000000000000000000000001006;
+  address internal constant _PLEDGE_AGENT_ADDR = 0x0000000000000000000000000000000000001007;
+  address internal constant _BURN_ADDR = 0x0000000000000000000000000000000000001008;
+  address internal constant _FOUNDATION_ADDR = 0x0000000000000000000000000000000000001009;
 
-  address public constant VALIDATOR_CONTRACT_ADDR = 0x0000000000000000000000000000000000001000;
-  address public constant SLASH_CONTRACT_ADDR = 0x0000000000000000000000000000000000001001;
-  address public constant SYSTEM_REWARD_ADDR = 0x0000000000000000000000000000000000001002;
-  address public constant LIGHT_CLIENT_ADDR = 0x0000000000000000000000000000000000001003;
-  address public constant RELAYER_HUB_ADDR = 0x0000000000000000000000000000000000001004;
-  address public constant CANDIDATE_HUB_ADDR = 0x0000000000000000000000000000000000001005;
-  address public constant GOV_HUB_ADDR = 0x0000000000000000000000000000000000001006;
-  address public constant PLEDGE_AGENT_ADDR = 0x0000000000000000000000000000000000001007;
-  address public constant BURN_ADDR = 0x0000000000000000000000000000000000001008;
-  address public constant FOUNDATION_ADDR = 0x0000000000000000000000000000000000001009;
+  uint public constant CORE_MAINNET = 1116;
+  uint public constant CORE_TESTNET = 1115;
+  uint public constant ANVIL_CHAINID = 31337;
+  uint public constant GANACHE_CHAINID = 1337;
 
+  // ReentrancyGuard
+  uint256 private constant GUARD_NOT_ENTERED = 0;
+  uint256 private constant GUARD_ENTERED = 100;
 
-  modifier onlyCoinbase() {
-  
-    require(msg.sender == block.coinbase, "the message sender must be the block producer");
-  
+  address immutable public s_testModeDeployer;
```

Comment:
immutable so not on storage

Reply:
To be included in the aforementioned 'storage modification guide'